File: src/bin/myecho.c
#include <stdio.h>

int main(int argc, int *argv[]){
    for(size_t i = 1; i < argc; i++){
        printf("%s ", argv[i]);
    }
    printf("\n");
}

// starting with i = 1, otherwise would've started with the execution command: ./myecho = argv[0]

File: src/bin/myecho.rs
use std::env;

fn main(){
    let args: Vec<String> = env::args().collect();

    for args in &args[1..] {
        print!("{} ", args);
    }
    println!();
}

// starting with i = 1, otherwise would've started with the execution command: target/debug/myecho = argv[0]